### PR TITLE
Implement mounting and decrypting

### PR DESCRIPTION
**Pull request by ppp0**

### CODE REVIEW

**Comment by tomaszdurka** [comment]
Changes above restructure code a bit.
Still to implement *comparison of filesystems*.

File: library/S3Export/library/S3Export/Cli.php [hunk 1]
```diff
@@ -23,8 +23,26 @@ public function getStatus($jobId) {
...
+        if ($device->getPartitionCount() === 1) {
+            $device->fixPartitioning();
+        }
+        $device->mount();
+        $truecryptImageFile = \Functional\first($device->getMountpoint()->listFiles(), function (CM_File $file) {
+            return $file->getExtension() === 'tc';
+        });
+        if (null === $truecryptImageFile) {
+            throw new CM_Cli_Exception_Internal("Cannot find truecrypt image on {$device->getPath()}");
+        }
```
Comment (tomaszdurka) [inline library/S3Export/library/S3Export/Cli.php]: @njam I was looking for assert framework for filesystems integrity check and looked into php assert more deeply. Is there any downside of using it in cases like above? Only downside is the message would be less guiding.
What I mean is converting:
```php
if (null === $truecryptImageFile) {
    throw new CM_Cli_Exception_Internal("Cannot find truecrypt image on {$device->getPath()}");
}
```
to
```php
assert($truecryptImageFile instanceof CM_File);
```

Comment (njam) [inline library/S3Export/library/S3Export/Cli.php]: > Assertions should be used as a debugging feature only. You may use them for sanity-checks that test for conditions that should always be TRUE and that indicate some programming errors if not or to check for the presence of certain features like extension functions or certain system limits and features.

> Assertions should not be used for normal runtime operations like input parameter checks. As a rule of thumb your code should always be able to work correctly if assertion checking is not activated.
http://php.net/manual/en/function.assert.php

The behaviour of `assert()` can be changed by configuration. I think it's meant for the such use cases.

**Comment by tomaszdurka** [comment]
Paired with @ppp0

**Pull request merged**